packages/react-tooltip/Spec.md

```diff
@@ -1,63 +1,515 @@
+
+export { TooltipProps };
+
+export const tooltipShorthandProps = [] as const;
+export const tooltipSlotProps = [...tooltipShorthandProps, 'arrow'] as const;
+
+export type TooltipState = ComponentState<
+  TooltipProps & {
+    /**
+     * The arrow that points to the target element.
+     */
+    arrow?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement> & React.RefAttributes<HTMLElement>>;
+  },
+  /* ShorthandProps: */ typeof tooltipShorthandProps[number],
+  /* DefaultedProps: */ 'arrow' | 'placement'
+>;
+```
+
+### TooltipProvider
+
+`TooltipProvider` is responsible for providing the actual implementation of `TooltipManagerApi` and `Tooltip`. It uses a React context to that contains those when it is included in the tree. This context will also be built into `FluentContext`.
+
+```typescript
+export interface TooltipProviderProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  // TooltipProvider has no additional props
+}
+
+export const tooltipProviderShorthandProps = [] as const;
+
+export type TooltipProviderState = ComponentState<TooltipProviderProps, typeof tooltipProviderShorthandProps[number]>;
+```
+
+### TooltipManager
+
+`TooltipManager` handles showing and hiding tooltips, including positioning and tracking the visible tooltip. There is only one instance at the root of the app.
+
+Note that the `TooltipManager` has a slot for the visible tooltip on its State, but it's not a public prop. Instead, the tooltip is shown/hidden using the `TooltipManagerApi` imperative interface, which is defined in the `react-tooltip-hook` package.
+
+> **Review:** TooltipProvider and TooltipManager could probably be merged into a single component
+
+```typescript
+import { TooltipManagerApi } from '@fluentui/react-tooltip-hook';
+
+export interface TooltipManagerProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * Ref to the imperative interface to show and hide tooltips.
+   */
+  componentRef?: React.Ref<TooltipManagerApi>;
+}
+
+export const tooltipManagerShorthandProps = ['tooltip'] as const;
+
+export type TooltipManagerState = ComponentState<
+  TooltipManagerProps & {
+    /**
+     * The Tooltip being rendered, if any.
+     */
+    tooltip?: ShorthandProps<TooltipProps>;
+  }
+>;
+```
+
+# Structure
+
+## Public
+
+```tsx
+<TooltipProvider>
+  <Button tooltip="Example tooltip" />
+  <TooltipTrigger tooltip="Element wrapped with TooltipTrigger">
+    <a href="http://example.com" />
+  </TooltipTrigger>
+</TooltipProvider>
+```
+
+## DOM
+
+In this example, the mouse is hovering over the first tooltip in the example above
+
+```tsx
+<body>
+  <div {/* TooltipProvider */}>
+    <button aria-describedby="tooltip-1" onPointerEnter={...} onPointerLeave={...} onFocus={...} onBlur={...} />
+    <a href="http://example.com" aria-describedby="tooltip-2" onPointerEnter={...} onPointerLeave={...} onFocus={...} onBlur={...} />
+  </div>
+
+  <div {/* Tooltip portal */}>
```

Comment:
Is the portal exclusive for tooltips or will it be shared with `Popups`? Which component owns it? It should be `ThemeProvider` as it needs to inject a theme as CSS variables.

Reply:
It doesn't necessarily need to be shared by Popup. We do need to ensure that tooltips are always layered above popups; not sure how easy that is if they share the same portal?

I was thinking the portal would be owned by `TooltipProvider`; and the provider would need to be under a `ThemeProvider` for theming support.